codegen/tests/two_worlds.rs

```diff
@@ -46,16 +46,16 @@ async fn main() {
         .run("./tests/features")
         .await;
 
-    assert_eq!(writer.steps.passed, 7);
-    assert_eq!(writer.steps.skipped, 5);
-    assert_eq!(writer.steps.failed, 0);
+    assert_eq!(writer.left.steps.passed, 7);
+    assert_eq!(writer.left.steps.skipped, 5);
+    assert_eq!(writer.left.steps.failed, 0);
```

Comment:
@ilslv hmmm... not very nice to have `.left`/`.right` without explicit `Or` usage in trivial cases.

Reply:
Also, we've discussed in voice the reasoning behind `writer::Or`.

Reply:
@tyranron I see 3 ways we can deal with `writer::Or`:

1. Leave it as is: `Cucumber<..., writer::Or<..., ...>>`
2. Storing it as `Box<dyn Writer>`. The problem with this approach, that we should store `Box<dyn Writer + writer::Failure + ...>` to provide full capabilities, but without specialisation we can't know what traits where implemented for a  original `Writer`, before it was wrapped into `writer::Or`
3. Returning user `Either<Cucumber<..., Writer1>, Cucumber<..., Writer2>>`, which gives most capabilities, but seems very unpractical to me. 

The main problem is, that we can't know a type based on a runtime value (in our case it's `--format=json` CLI argument), so I would pick a first option, as it most ergonomic and should be enough for most of the cases. And in case you need more capabilities, option 3 can be achieved with manual impl: 
```rust
if cli.format {
    Either::Left(
        World::cucumber()
            .writer(...)
    )
} else {
    Either::Right(
        World::cucumber()
            .writer(...)
    )
}
```

Reply:
@ilslv ok, then let's hide struct fileds behind methods, to make `writer::Or` not breaking existing code.